server/src/lib.rs

```diff
@@ -218,7 +218,22 @@ fn fetch_iframely_and_pictshare_data(
         None
       }
     },
-    None => None,
+
+    None => match url {
+      Some(url) => match fetch_pictshare(&url) {
```

Comment:
How do you know this url is a picture?

Reply:
We don't know, but if it is, pictshare will return an "ok" status.

Reply:
I'm okay with that I spose.

Reply:
Also why are you matching the url a second time? You already know that its some.

Reply:
In this match arm, we don't know if it's some. iframely_thumbnail_url is None when iframely raises an error or when url is None. As a result, we have to recheck if url is None. This can be avoided by rewriting the function but I thought that would be outside the scope of the PR. I can rewrite the function in another branch.